### DISCUSSION TITLE
Gallery

### DISCUSSION THREAD

**Discussion by pthom** [pthom/imgui_bundle#107] [Show and tell]
This discussion is dedicated to users posting example of how they use Dear ImGui Bundle. I'm eager to see your applications and how you've integrated it into your projects.

Additionally, please share your examples in the Dear ImGui Galleries as well if they might interest a broader audience. You can find the relevant gallery in the [Dear ImGui repository](https://github.com/ocornut/imgui/issues) under "issues".

See related [gallery for Hello ImGui](https://github.com/pthom/hello_imgui/discussions/82)

**Comment by pthom** [comment]
The interactive manual for Dear ImGui Bundle is of course a good and quite complete demonstration.

<img width="1166" alt="image" src="https://user-images.githubusercontent.com/7694091/236272526-9d716e30-410b-4388-9094-f21d4ea8b9f8.png">

It is [available online](https://traineq.org/ImGuiBundle/emscripten/bin/demo_imgui_bundle.html).

**Comment by pthom** [comment]
[Dear ImGui Manual](https://pthom.github.io/imgui_manual_online/manual/imgui_manual.html) is an interactive online manual for Dear ImGui.

<img width="1211" alt="image" src="https://user-images.githubusercontent.com/7694091/236273726-86ffa9bf-f304-4915-9e6e-5874cf662443.png">

**Comment by pthom** [comment]
[Online demo for Implot](https://traineq.org/implot_demo/src/implot_demo.html) (with example code browsing)
<img width="1118" alt="image" src="https://user-images.githubusercontent.com/7694091/236274333-f78eaf47-7798-4ea6-9a07-240ad5dbf337.png">

**Comment by pthom** [comment]
fiatlux : Work in progress on a generic way to display function pipelines and to visually examine and debug intermediate states. Not public for the moment.

Example with an image processing pipeline:
<img width="1595" alt="image" src="https://user-images.githubusercontent.com/7694091/236275790-a7370197-f394-4406-a15c-ef26f9d71bca.png">

Example with a text processing pipeline (calculate word frequencies):
<img width="1275" alt="image" src="https://user-images.githubusercontent.com/7694091/236276694-1a720342-6e07-4f54-8e94-bfb6c883dee0.png">

Example with a basic math pipeline:
<img width="757" alt="image" src="https://user-images.githubusercontent.com/7694091/236277333-a51dddbc-ec79-4ec1-a306-df07483cc3d8.png">

**Reply by dcnieho** [reply]
This is really cool. Please do post an update when its ready!

**Comment by dcnieho** [comment]
File commander-style interface that can browse local and remote machines, part of [labManager](https://github.com/dcnieho/labManager/).
Work in progress (readme is missing). LabManager is a master-client system for remote control and file management of machine(s) in a computer lab, with an optional component for disk image management. Only possible thanks to Pascal's great work with comprehensively wrapping an up-to-date version of imgui, and making it easy to use with the hello imgui framework!

![image](https://github.com/pthom/imgui_bundle/assets/1787673/325145eb-4e62-405c-9d12-4923f6039364)

**Comment by pthom** [comment]
[ 4K4D: Real-Time 4D View Synthesis at 4K Resolution](https://zju3dv.github.io/4k4d/) 

https://private-user-images.githubusercontent.com/43734697/288741193-faf60953-fd7f-4309-a1f5-758eaa1182ca.mp4?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI1ODc5NzAsIm5iZiI6MTcxMjU4NzY3MCwicGF0aCI6Ii80MzczNDY5Ny8yODg3NDExOTMtZmFmNjA5NTMtZmQ3Zi00MzA5LWExZjUtNzU4ZWFhMTE4MmNhLm1wND9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDA4VDE0NDc1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY2YThiM2FlNWUxOTdmZTQ5Yzg4MDMzNDZmM2M2ZGE0MTNhNTE5NDcyM2FhNjJmODY5ODBjN2ZiZjUwNGNkOTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.KZ4Hc0Vc9mXAcfPWti_QE9qbJRWSFImrE6HEd_ym7T0


https://private-user-images.githubusercontent.com/43734697/288741154-379b2793-eda2-4720-a1ad-cbfd2205af4d.mp4?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI1ODgxOTgsIm5iZiI6MTcxMjU4Nzg5OCwicGF0aCI6Ii80MzczNDY5Ny8yODg3NDExNTQtMzc5YjI3OTMtZWRhMi00NzIwLWExYWQtY2JmZDIyMDVhZjRkLm1wND9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDA4VDE0NTEzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFiMjk3NTU0YjE0N2NmOThiOGU1MWYzN2ZlMGE1YTgzNGRlYzk4YTk2ZGUwYmYzZDQxNWRiYjllZWYzMzMyNzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.cPoimIIl9w1yKCKGywAVGGG8v6J-NUlLxLq_DMpNvJI



A research project aimed for CVPR 2024, using python bindings (ImGui Bundle).
```
@inproceedings{xu20244k4d,
  title={4K4D: Real-Time 4D View Synthesis at 4K Resolution},
  author={Xu, Zhen and Peng, Sida and Lin, Haotong and He, Guangzhao and Sun, Jiaming and Shen, Yujun and Bao, Hujun and Zhou, Xiaowei},
  booktitle={CVPR},
  year={2024}
}
```

**Comment by Johndeep** [comment]
Currently a wip project, but something I yet to see implemented within pure Python. A simple transfer function tool.
![wip_transfer_fct](https://github.com/pthom/imgui_bundle/assets/306953/8338d360-5157-4390-8ca5-741e3031c289)
It has the ability to create custom alpha blending within colormaps from matplotlib. Or you can also create your own colormap by hand. This was implemented within glfw first, but now it is running within hello_imgui. (As of now, with Ubuntu 24.04, the glfw and sdl2 backend seems to have context error within Wayland, fortunately immapp and hello_imgui works fine).

**Reply by pthom** [reply]
Cool, thanks for sharing! I like to see what users are doing. Please post a link to your projects if/when you have one. 

> As of now, with Ubuntu 24.04, the glfw and sdl2 backend seems to have context error within Wayland, fortunately immapp and hello_imgui works fine

Do you mean it was not working when using ImGui alone, and it works when using Dear ImGui Bundle?

**Reply by Johndeep** [reply]
I have no online repo for this yet.

About the pyimgui and imgui_bundle situation. Since beginning of this year, I could not run pyimgui with glfw on Ubuntu with Wayland anymore. I think I need to tinker around a bit more to determine if it is a "me" problem, which can be fixed. But even then...

... about the imgui_bundle situation. If you just run the Python sdl2 and glfw backend example in 24.04 in Wayland, you'll currently get nasty context error. Pure sdl2 and pure glfw works fine in Wayland. But as soon as I add this line for example:
`impl = glfw_backend.GlfwRenderer(window)`
I immediately get context error. I hope this is not a "me" problem. But fortunately, It seems that hello_imgui and immapp both are working just fine, as you can see in the screenshot above. Well, due to this transition, my app seems to be buggy again here and there..
